src/renderer/components/dock/editor-panel.tsx

```diff
@@ -59,7 +61,7 @@ const NonInjectedEditorPanel = observer(({
       autoFocus={autoFocus}
       id={tabId}
       value={value}
-      className={cssNames(styles.EditorPanel, className)}
+      className={cssNames(styles.EditorPanel, className, { hidden })}
```

Comment:
This seems strange, why is this needed?

Reply:
Because `MonacoEditor` works incorrectly

```typescript
const SomeComponent = observer(({ someProp, value}: Props) => (
  <div>
    {someProp.get() ? <MonacoEditor value={value.get()} .../> : <div>Something else</div>}
  </div>
));

If you were to change `value` when `someProp` is falsy (`MonacoEditor` is no rendered), the value of `MonacoEditor` would be incorrect when `someProp` changes to truthy.

Reply:
Then shouldn't we fix `MonacoEditor`? That sounds like it isn't checking for an updated `value` prop (in `componentDidUpdate` or something).

Reply:
Yes, we should. Draw the line here, because there is lot to improve in `MonacoEditor`. Issue is that it's calling `this.restoreViewState(this.model);` when it creates the editor. Didn't go any deeper than that.